internal/controller/pkg/manager/reconciler.go

```diff
@@ -497,12 +523,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	pr.SetSkipDependencyResolution(p.GetSkipDependencyResolution())
 	pr.SetCommonLabels(p.GetCommonLabels())
 
-	pwr, pwok := p.(v1.PackageWithRuntime)
-	prwr, prok := pr.(v1.PackageRevisionWithRuntime)
-	if pwok && prok {
-		prwr.SetRuntimeConfigRef(pwr.GetRuntimeConfigRef())
-		prwr.SetTLSServerSecretName(pwr.GetTLSServerSecretName())
-		prwr.SetTLSClientSecretName(pwr.GetTLSClientSecretName())
+	if r.setPackageRuntimeSecrets != nil {
```

Comment:
IIUC, after this changes, external runtimes will now need to reconcile Package types as well. Previously, reconciling only PackageRevisions was sufficient, but not, they would at least need to propagate runtimeConfigRef from package to revision and initialize the TLS secret names, right?

Which sounds reasonable, it is a runtime concern and XP package manager should not do anything in that regard when an external runtime is configured.

Reply:
There is indeed one other usage of this TLS secrets in the revision reconcilier which is unfortunately a little convoluted. While deploying the CRDs, we need to inject tls.crt as webhook tls secret, which makes it indeed a little shared between package establisment and runtime. Currently, runtime generates this secret and revision reconciler consumes it. Not sure if this PR would change anything there, but at least here, we may want to use observed tls secret: https://github.com/crossplane/crossplane/blob/106493b50b1209e8f434123389735f3c12e96eb8/internal/controller/pkg/revision/establisher.go#L372